tgui/packages/tgui/interfaces/GeneTek/modals/BuyMaterialsModal.tsx

```diff
@@ -17,7 +17,7 @@ export const BuyMaterialsModal = (props) => {
   const maxBuyMats = props.maxAmount;
   const { budget, costPerMaterial } = data;
 
-  const resolvedBuyMats = Math.min(buyMats, maxBuyMats);
+  const resolvedBuyMats = buyMats !== null ? Math.min(buyMats, maxBuyMats) : 1;
```

Comment:
Can't easily leave as a diff comment, but:

Set `buyMats` to default to `1` instead of `null` if that's what's going to be used anyway, then the `useSharedState` would infer `number` and this line wouldn't need changing.

Reply:
The problem is that, in `GeneTek.tsx`, the buyMats state is also used to determine whether the popup is open or not,

Later on in BuyMaterialsModal, buyMats is set to null to close the popup.

Oh! Just had the idea of using 0 instead of null, that should solve everything.